### ISSUE TITLE
Scrambled screen after scrolling

### ISSUE THREAD

**Issue by erka2020** [wesnoth/wesnoth#4694]
### Game and System Information
Windows10 Home, 1909 
ATI RX570 Graphics adapter Version Driver Version 19.20
Battle for Wesnoth v1.14.9 (9eba78f-Modified) (Steam version)

The Battle for Wesnoth version 1.14.9 (9eba78f-Modified)
Running on Microsoft Windows 10 (10.0.18363)

Game paths
==========
Data dir:        E:\Steam\steamapps\common\wesnoth
User config dir: C:\Users\USER\Documents\My Games\Wesnoth1.14
User data dir:   C:\Users\USER\Documents\My Games\Wesnoth1.14
Saves dir:       C:\Users\USER\Documents\My Games\Wesnoth1.14\saves
Add-ons dir:     C:\Users\USER\Documents\My Games\Wesnoth1.14\data\add-ons
Cache dir:       C:\Users\USER\Documents\My Games\Wesnoth1.14\cache

Libraries
=========
Boost:             1.66
OpenSSL/libcrypto: 1.1.0f (runtime 1.1.0f)
Cairo:             1.10.2 (runtime 1.10.2)
Pango:             1.30.1 (runtime 1.30.1)
SDL:               2.0.8  (runtime 2.0.8)
SDL_image:         2.0.3  (runtime 2.0.3)
SDL_mixer:         2.0.2  (runtime 2.0.2)
SDL_ttf:           2.0.14 (runtime 2.0.14)

### Describe the bug
After playing for about 2 hours the screen suddenly scrambles when scrolling, reproducing tiles over and over again. Moving the cursor ober a unit restores the screen. Scrolling again reproduces the error. Restarting the game with the same savegame does not reproduce the error at once. The error reoccurs after a new playing time of about one to two hours.

### To Reproduce
Play wesnoth for about 2-3 hours. 

### Screenshots
![Scrambled Screen](https://user-images.githubusercontent.com/59666861/72019562-3e37fc00-326a-11ea-9c45-d2603e2ee41e.jpg)

**Comment by erka2020** [comment]
Update: This time short after the first scrambled screen, wesnoth crashes with a minGW error
Assertion failed
![Error Msg](https://user-images.githubusercontent.com/59666861/72087206-50687780-3308-11ea-9f30-e38b63674abf.jpg)

**Comment by shikadiqueen** [comment]
Can you verify that when this happens your graphics card is not overheating when this occurs? Also, make sure that you are using the latest drivers from the manufacturer for it.

**Comment by GregoryLundberg** [comment]
The assertion failure would indicate there is no destination structure to receive a graphics element, although I must admit I don't see why if could ever fail since it's a pass-by-reference.

**Comment by erka2020** [comment]
According to MSI Afterburner the card temperature is about 55°.
Driver software was updated when I opened the issue, but on 9/1/2020 a new driver version was released. I just installed it and keep on playing.

**Comment by Wedge009** [comment]
Bugs appearing over time are hard to diagnose and solve. Is the time to fail consistent? This sort of reminds me of #4447 (where the refreshing accelerates whatever process, memory leak, or whatever, causing the graphical artefacts).

**Comment by erka2020** [comment]
The time to fail is not consistent - but high (making the problem less painful). It usually takes rather hours then minutes to occur and a correlation with repeately reloading savegames is possible.
(Finally the crappy Windows 10 drivers from AMD cannot be ruled out as well)

**Comment by Wedge009** [comment]
I doubt the issue is directly related to GPU since all rendering in Wesnoth is still CPU-based. But it's non-zero possibility, I suppose.

**Comment by yuhasem** [comment]
I ran into the same bug today.

Game and System Information
Windows 10 Home (19042.804)
Intel(R) UHD Graphics 630 driver version 26.20.100.6911

The Battle for Wesnoth version 1.14.12
Running on Microsoft Windows 10 (10.0.19042)
Distribution channel: SourceForge

Game paths
==========

Data dir:        C:\Program Files (x86)\Battle for Wesnoth 1.14.12
User config dir: C:\Users\hando\Documents\My Games\Wesnoth1.14
User data dir:   C:\Users\hando\Documents\My Games\Wesnoth1.14
Saves dir:       C:\Users\hando\Documents\My Games\Wesnoth1.14\saves
Add-ons dir:     C:\Users\hando\Documents\My Games\Wesnoth1.14\data\add-ons
Cache dir:       C:\Users\hando\Documents\My Games\Wesnoth1.14\cache

Libraries
=========

Boost:             1.59
OpenSSL/libcrypto: 1.1.0f (runtime 1.1.0f)
Cairo:             1.10.2 (runtime 1.10.2)
Pango:             1.30.1 (runtime 1.30.1)
SDL:               2.0.12 (runtime 2.0.12)
SDL_image:         2.0.5  (runtime 2.0.5)
SDL_mixer:         2.0.4  (runtime 2.0.4)
SDL_ttf:           2.0.15 (runtime 2.0.15)

For me the bug started after about 2 hours of gameplay and crashed within a few minutes.  I've played this long before without issue, so I don't have a regularly reproducible case.  There were these error messages in the logs just before the crash [wesnoth-20210307-090935-9604.log](https://github.com/wesnoth/wesnoth/files/6098020/wesnoth-20210307-090935-9604.log)

Failed to create a surface of size 1098x930 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 1098x930 Reason: Out of memory
20210307 11:03:03 error gui/draw: SDL_CreateSoftwareRenderer failed, reason: Can't create renderer for NULL surface
Failed to create a surface of size 1098x930 Reason: Out of memory
20210307 11:03:03 error gui/draw: SDL_CreateSoftwareRenderer failed, reason: Can't create renderer for NULL surface
